Core/HLE/sceNetAdhoc.cpp

```diff
@@ -1107,62 +1287,94 @@ static u32 sceNetAdhocctlDisconnect() {
 	return 0; //ERROR_NET_ADHOC_NOT_INITIALIZED; // Wipeout Pulse will repeatedly calling this function if returned value is ERROR_NET_ADHOC_NOT_INITIALIZED
 }
 
+static u32 sceNetAdhocctlDisconnect() {
+	// WLAN might be disabled in the middle of successfull multiplayer, but we still need to cleanup right?
+	char grpName[9] = { 0 };
+	memcpy(grpName, parameter.group_name.data, ADHOCCTL_GROUPNAME_LEN); // Copied to null-terminated var to prevent unexpected behaviour on Logs
+	INFO_LOG(SCENET, "sceNetAdhocctlDisconnect() at %08x [group=%s]", currentMIPS->pc, grpName);
+
+	return NetAdhocctl_Disconnect();
+}
+
 static u32 sceNetAdhocctlDelHandler(u32 handlerID) {
 	if (adhocctlHandlers.find(handlerID) != adhocctlHandlers.end()) {
 		adhocctlHandlers.erase(handlerID);
-		WARN_LOG(SCENET, "UNTESTED sceNetAdhocctlDelHandler(%d): deleted handler %d", handlerID, handlerID);
+		WARN_LOG(SCENET, "UNTESTED sceNetAdhocctlDelHandler(%d)", handlerID);
 	} else {
-		ERROR_LOG(SCENET, "UNTESTED sceNetAdhocctlDelHandler(%d): asked to delete invalid handler %d", handlerID, handlerID);
+		ERROR_LOG(SCENET, "UNTESTED sceNetAdhocctlDelHandler(%d): Invalid Handler ID", handlerID);
 	}
 
 	return 0;
 }
 
-int sceNetAdhocctlTerm() {
-	// WLAN might be disabled in the middle of successfull multiplayer, but we still need to cleanup right?
-	INFO_LOG(SCENET, "sceNetAdhocctlTerm()");
-
-	if (netAdhocMatchingInited) sceNetAdhocMatchingTerm();
-
+int NetAdhocctl_Term() {
 	if (netAdhocctlInited) {
-		netAdhocctlInited = false;
+		if (threadStatus != ADHOCCTL_STATE_DISCONNECTED) 
+			NetAdhocctl_Disconnect();
+
+		// Terminate Adhoc Threads
 		friendFinderRunning = false;
 		if (friendFinderThread.joinable()) {
 			friendFinderThread.join();
 		}
+
+		// Clear Peer List
+		int32_t peercount = 0;
+		freeFriendsRecursive(friends, &peercount);
+		INFO_LOG(SCENET, "Cleared Peer List (%i)", peercount);
+		// Delete Peer Reference
+		friends = NULL;
 		//May also need to clear Handlers
 		adhocctlHandlers.clear();
```

Comment:
will be my next note on testing with auto test does term really removing the adhocctl handler?